### PR TITLE
Us1 contact coord

### PR DESCRIPTION
**Pull request by CeliaBoiss**

### CODE REVIEW

**Review by thocle86 — DISMISSED** [review]
ok

**Review by Quentin-BLANC — DISMISSED** [review]

**Review by FLorianBlm — DISMISSED** [review]

**Review by Quentin-BLANC — APPROVED** [review]

**Review by Quentin-BLANC — APPROVED** [review]

**Review by Quentin-BLANC — DISMISSED** [review]

**Review by Quentin-BLANC — DISMISSED** [review]

**Review by FLorianBlm — DISMISSED** [review]

**Review by Quentin-BLANC — DISMISSED** [review]

**Review by thocle86 — DISMISSED** [review]
ok

File: src/FormData/MailMessageData.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+     * @Assert\Type(type="string")
+     */
+    private $phone;
+
+    /**
+     * @var string
+     * @Assert\NotBlank(message="Veuillez saisir votre email")
+     * @Assert\Type(type="string")
+     */
+    private $email;
```
Comment (RomainClair) [inline src/FormData/MailMessageData.php]: Symfony has an email constraints : https://symfony.com/doc/current/reference/constraints/Email.html

File: src/FormData/MailMessageData.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+     * @Assert\Type(type="string")
+     */
+    private $subject;
+
+    /**
+     * @var string
+     * @Assert\NotBlank(message="Veuillez saisir votre numéro de téléphone")
+     * @Assert\Type(type="string")
+     */
+    private $phone;
```
Comment (RomainClair) [inline src/FormData/MailMessageData.php]: You should probably check to large reasonnable maximum size

File: templates/emails/contact.html.twig [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<ul>
+    <li>Sujet : {{ message.subject }}</li>
```
Comment (RomainClair) [inline templates/emails/contact.html.twig:2]: What's the point of repeating the subject and the mail, they already are in in the mail headers !

**Review by RomainClair — CHANGES_REQUESTED** [review]
The HomeController::sendMail() private method could reasonnably be integrated in the index method

You should use twig includes when you need them.
If you don't plan to reuse the part elsewhere, it probably should be in the templates.

**Review by Quentin-BLANC — DISMISSED** [review]

**Review by Quentin-BLANC — APPROVED** [review]

**Review by FLorianBlm — APPROVED** [review]

**Review by thocle86 — APPROVED** [review]

File: src/FormData/MailMessageData.php [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+     * @Assert\Type(type="string")
+     * @Assert\Length(max="25")
+     */
+    private $phone;
+
+    /**
+     * @var string
+     * @Assert\NotBlank(message="Veuillez saisir votre email")
+     * @Assert\Email
+     * @Assert\Length(max="100")
```
Comment (RomainClair) [inline src/FormData/MailMessageData.php:29]: There may be valid email addresses with more than 100 caracters. A quick search tells It can be up to 254

**Review by RomainClair — APPROVED** [review]
You're length constraints are probably too small.
Firstname & lastnames can be longer than 50 for example.

**Pull request merged**